test/end-to-end/test_pkg_uninstall.ps1

```diff
@@ -0,0 +1,57 @@
+$pkg = "core/redis"
+$pkgs = @(
+    "$pkg/3.2.3/20161102201135",
+    "$pkg/3.2.4/20161104175435",
+    "$pkg/3.2.4/20161210004233",
+    "$pkg/3.2.4/20161215055911",
+    "$pkg/3.2.4/20170103160441",
+    "$pkg/3.2.4/20170106011058",
+    "$pkg/4.0.10/20180801003001",
+    "$pkg/4.0.10/20190116005049",
+    "$pkg/4.0.14/20200319184753",
+    "$pkg/4.0.14/20200319200053"
+)
+
+Describe "pkg uninstall" {
+    It "installs core/redis" {
+        foreach ($p in $pkgs) {
+            hab pkg install "$p"
+        }
+        hab pkg list "$pkg" | Should -BeExactly $pkgs
+    }
+
+    It "uninstall a single package" {
+        hab pkg uninstall "$pkg"
+        hab pkg list "$pkg" | Should -BeExactly $pkgs[0..8]
+    }
+
+    It "uninstall all but the two latest of version 3.2.4" {
+        hab pkg uninstall --keep-latest=2 "$pkg/3.2.4"
+        hab pkg list "$pkg" | Should -BeExactly $pkgs[,0+4..8]
+    }
+
+    It "dry run should do nothing" {
+        hab pkg uninstall -d "$pkg"
+        hab pkg uninstall -d --keep-latest=1 "$pkg"
+        hab pkg list "$pkg" | Should -BeExactly $pkgs[,0+4..8]
+    }
+
+    It "uninstall all but the three latest" {
+        hab pkg uninstall --keep-latest=3 "$pkg"
+        hab pkg list "$pkg" | Should -BeExactly $pkgs[6..8]
+    }
+
+    It "uninstall does nothing if keeping all" {
+        hab pkg uninstall --keep-latest=10 "$pkg"
+        hab pkg list "$pkg" | Should -BeExactly $pkgs[6..8]
+    }
+
+    It "uninstall all" {
+        hab pkg uninstall --keep-latest=0 "$pkg"
```

Comment:
I thought `keep-latest` was meant to be a non-zero integer?

I'm also somewhat surprised we don't have `hab pkg uninstall --all $PKG`.

Perhaps somewhat at odds with my argumentation above, I wonder if we _should_ treat `--keep-latest=0` as an error, and instead implement `--all` as an explicit indication of "yes, I really mean to get rid of _everything_". Perhaps I'm overthinking, but I wonder about scripted use of this where the number given for `--keep-latest` is a variable that, due to some script bug, ends up accidentally becoming 0. In that situation, you just accidentally blew away all your installed instances of that package.

I'd love some of your thoughts on this UX, @apriofrost :smile:

Reply:
I'm assuming `--all` or `--keep-latest=0` is the default here if you do not provide a `--keep-latest`. Personally I am fine with allowing 0. I don't find it confusing or surprising. Just my 2 cents.

Reply:
No, it is not currently the default. If a partial ident is specified with no `--keep-latest` flag only the newest package is removed. Truthfully, I find this behavior a little surprising. If it was only going to remove one package I would have thought it would be the oldest.

I could see that it would make sense for `--keep-latest` to default to `0` and then rely on the user to use full or partial idents to specify exactly what they want to remove. For example, if someone ran `hab pkg uninstall core/redis`, I don't think it would be all that crazy to assume they want to remove all `core/redis` packages. But this would be a breaking change and a pretty dangerous one at that.

Reply:
@krasnow I am curious what your thoughts are here?

Reply:
`--keep-latest=0` seems like a good option to have. It follows that if I wanted to delete all packages, this would be a valid way to do so.

As David says above, there's a separate usability problem with the default behavior of `hab pkg unstall` in that I would expect that to delete all versions of a package if I don't specify an ident. We should split that out as a separate issue, but fixing that would make the `keep-latest` option more intuitive.

Reply:
#7604 is the separate issue for addressing `hab pkg uninstall` UX.